Sources/TokamakLint/Reporters/TokamakLogger.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  TokamakLogger.swift
+//  TokamakLint
+//
+//  Created by Matvii Hodovaniuk on 5/30/19.
+//
+
+import Foundation
+import Logging
+
+enum Output {
+  case stdout
+  case file
+}
+
+public struct Outputs: OptionSet {
+  public init(rawValue: Outputs.RawValue) {
+    self.rawValue = rawValue
+  }
+
+  public let rawValue: Int
+
+  static let stdout = Outputs(rawValue: 1)
+  static let file = Outputs(rawValue: 2)
+}
+
+public struct TokamakLogger: LogHandler {
+  private static var overrideLogLevel: Logger.Level?
+
+  // this holds the log level if not overridden
+  private var _logLevel: Logger.Level = .info
+
+  // metadata storage
+  public var metadata: Logger.Metadata = [:]
+
+  public init(label: String) {}
+
+  public var outputs = [Outputs.file]
+
+  public var _path: URL?
+
+  public var path: URL {
+    get {
+      guard let _path = _path else {
+        let fm = FileManager.default
+        guard let url = fm.urls(
+          for: FileManager.SearchPathDirectory.documentDirectory,
+          in: FileManager.SearchPathDomainMask.userDomainMask
+        )
+        .last?.appendingPathComponent("log.txt") else {
+          return URL(fileURLWithPath: "")
+        }
+        return url
+      }
+      return _path
+    }
+
+    set {
+      _path = newValue
+    }
+  }
+
+  public var logLevel: Logger.Level {
+    // when we get asked for the log level, we check if it was
+    // globally overridden or not
+    get {
+      return _logLevel
+    }
+    // we set the log level whenever we're asked (note: this might not
+    // have an effect if globally overridden)
+    set {
+      _logLevel = newValue
+    }
+  }
+
+  public func log(
+    level: Logger.Level,
+    message: Logger.Message,
+    metadata: Logger.Metadata?,
+    file: String,
+    function: String,
+    line: UInt
+  ) {
+    if outputs.contains(Outputs.stdout) {
+      print(message.description)
+    }
+
+    if outputs.contains(Outputs.file) {
+      write(message.description)
+    }
+  }
+
+  public func log(level: Logger.Level, message: Logger.Message) {
+    log(
+      level: level,
+      message: message,
+      metadata: .none,
+      file: "",
+      function: "",
+      line: 0
+    )
+  }
+
+  func write(_ string: String) {
+    let data = string.data(using: String.Encoding.utf8)
+    let fm = FileManager.default
+    if !fm.fileExists(atPath: path.absoluteString) {
+      fm.createFile(
+        atPath: path.absoluteString,
+        contents: data, attributes: nil
+      )
+    } else {
+      var file = FileHandle(forReadingAtPath: path.absoluteString)
+      _ = String(describing: file?.readDataToEndOfFile())
+      file = FileHandle(forWritingAtPath: path.absoluteString)
```

Comment:
it's unclear what this code is doing and why the filehandle is opened twice for every log message. I think it would more effective to open `FileHandle` once in the initializer and to reuse the handle across writes